.github/workflows/publish-docs.yaml

```diff
@@ -0,0 +1,51 @@
+name: Publish Docs
+
+on:
+  workflow_call:
+    inputs:
+      tag:
+        description: 'Release tag'
+        required: true
+        type: string
+
+jobs:
+  publish-docs:
+    name: Publish Docs ${{ inputs.tag }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout Foundry Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Checkout GitBook Repository
+        uses: actions/checkout@v4
+        with:
+          repository: pantos-io/docs
+          path: ./pantos-docs
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+
+      - name: Generate ABI Documentation
+        run: |
+          make docs-abis
+
+      - name: Copy ABI Documentation
+        run: |
+          cp docs/abis/pantos-hub-abi.md ./pantos-docs/general/pantoshub-abi.md
+          cp docs/abis/pantos-token-abi.md ./pantos-docs/general/pantostoken-abi.md
+
+      - name: Update GitBook Repository
+        run: |
+          git config --global user.name "Don Pantos"
```

Comment:
Have you tested it? I believe it won't work properly.
You checkout the Ethereum repo and into it the docs repo, so you have a repo in a repo.

It will probably push to Ethereum-contracts instead of docs.

Also not sure if the GitHub token has the permissions to push to docs main.

Reply:
A tested it here in the PR as much as I could. The push was excluded from that, but a similar approach works here:
https://github.com/pantos-io/cd-workflows/blob/main/.github/workflows/update_addresses_at_docs.yml

Reply:
Let's try, I am myself curious :) I suspect it won't work, but we can just fix it after the first run :)